### ISSUE TITLE
[Bug]: Gives 403 Error When Try to Download from Udem Personal Subscripton Courser

### ISSUE THREAD

**Issue by puffin-000** [Puyodead1/udemy-downloader#247]
### What happened?

The script works when I try to download from the courses I bought exclusively, but failed to download courses from Udemy Personal plan. I used "--browser brave" and "-sc" arguments. They don't work, what can I do about that? 

### Expected Result

[udemy-downloader] [main:1843] INFO: > Fetching course information, this may take a minute...
[udemy-downloader] [main:1846] INFO: > Course information retrieved!
[udemy-downloader] [main:1851] INFO: > Fetching course curriculum, this may take a minute...
[udemy-downloader] [main:1867] INFO: > Course curriculum retrieved!
[udemy-downloader] [main:1894] INFO: > Processing course data, this may take a minute.

### Branch

master/main

### What operating systems are you seeing the problem on?

Linux/Unix

### Relevant log output

```shell
[10:56:31] [udemy-downloader] [pre_run:331] INFO: Output directory set to /home/puffin/udl/out_dir
[10:56:31] [udemy-downloader] [main:1843] INFO: > Fetching course information, this may take a minute...
[10:56:32] [udemy-downloader] [_get:1125] ERROR: Failed request https://www.udemy.com/course/python-sifirdan-ileri-seviyeye/learn
[10:56:32] [udemy-downloader] [_get:1126] ERROR: 403 Forbidden, retrying (attempt 0 )...
[10:56:34] [udemy-downloader] [_get:1125] ERROR: Failed request https://www.udemy.com/course/python-sifirdan-ileri-seviyeye/learn
[10:56:34] [udemy-downloader] [_get:1126] ERROR: 403 Forbidden, retrying (attempt 1 )...
[10:56:36] [udemy-downloader] [_get:1125] ERROR: Failed request https://www.udemy.com/course/python-sifirdan-ileri-seviyeye/learn
[10:56:36] [udemy-downloader] [_get:1126] ERROR: 403 Forbidden, retrying (attempt 2 )...
```

**Comment by Puyodead1** [comment]
provide the full command you are using. 
ensure you are logged in to udemy in brave. 
try removing the `/learn` part of the url

**Comment by puffin-000** [comment]
Command used: `python3 main.py --subscription-course -c https://www.udemy.com/course/c-temel-bilgiler-ve-uygulamal-egitim-kursu/ --browser brave`

I tried without learn and it gives this error now;
`[11:36:03] [udemy-downloader] [pre_run:331] INFO: Output directory set to /home/puffin/udl/out_dir
[11:36:03] [udemy-downloader] [main:1843] INFO: > Fetching course information, this may take a minute...
[11:36:06] [udemy-downloader] [_extract_subscription_course_info:940] CRITICAL: Could not find course data. Possible causes are: Missing cookies.txt file, incorrect url (should end with /learn), not logged in to udemy in specified browser.`

**Comment by puffin-000** [comment]
Is there a way to manually place cookie file? Maybe script can't get because I'm on linux. I guess its not fully tested like Windows.

**Comment by swahpy** [comment]
> Is there a way to manually place cookie file? Maybe script can't get because I'm on linux. I guess its not fully tested like Windows.

Yes it is possible. Just change `brave` to `file` and place your cookies in cookies.txt file. Actually this part is a yt-dlp stuff. You could check that in there.

**Comment by Puyodead1** [comment]
> > Is there a way to manually place cookie file? Maybe script can't get because I'm on linux. I guess its not fully tested like Windows.
> 
> Yes it is possible. Just change `brave` to `file` and place your cookies in cookies.txt file. Actually this part is a yt-dlp stuff. You could check that in there.

we talked in discord, they are getting cloudflare challenges

**Comment by puffin-000** [comment]
Cloudflare dropped the captcha challanges, I can download normally now. I'm closing this issue, but there should be a feature tracking for cloudflare bypass. Thanks for help.

**Comment by techshad** [comment]
> Cloudflare dropped the captcha challanges, I can download normally now. I'm closing this issue, but there should be a feature tracking for cloudflare bypass. Thanks for help.

sir can you tell me please how did you solve this problem, I am facing same problem of 403 forbidden request, the command which I am using is :python3 main.py -c https://www.udemy.com/course/the-complete-web-development-bootcamp/learn -n -q 1080 -l en -sc --download-assets --download-captions --download-quizzes --id-as-course-name -sc --browser brave

**Comment by puffin-000** [comment]
> > Cloudflare dropped the captcha challanges, I can download normally now. I'm closing this issue, but there should be a feature tracking for cloudflare bypass. Thanks for help.
> 
> sir can you tell me please how did you solve this problem, I am facing same problem of 403 forbidden request, the command which I am using is :python3 main.py -c https://www.udemy.com/course/the-complete-web-development-bootcamp/learn -n -q 1080 -l en -sc --download-assets --download-captions --download-quizzes --id-as-course-name -sc --browser brave

You used -sc variable two times, maybe that is the issue. About how I bypass Cloudflare, I actually didn't do anything. Cloudflare randomly stop giving challenges. Wait a couple day and try again, there is not certain way to bypass unless you are willing to buy captcha solver api.

**Comment by techshad** [comment]
thank you for your reply and time! the problem is solved now actually I got to know that I don't even need to mention -sc or -browser, just like normally we do for purchased courses, only the thing is in subscription account add that course to a list that's all
